kaiax/builder/impl/init.go

```diff
@@ -29,21 +34,31 @@ var (
 
 type InitOpts struct {
 	Backend api.Backend
+	TxPool  kaiax.TxPoolForCaller
+	Modules []builder.TxBundlingModule
+}
+
+type txAndTime struct {
+	tx   *types.Transaction
+	time time.Time
 }
 
 type BuilderModule struct {
 	InitOpts
+
+	pendingBundles map[common.Hash]txAndTime
```

Comment:
```suggestion
	knownTxs map[common.Hash]txAndTime
```

+ what do you think about introducing a new type `map[common.Hash]txAndTime`, so we don't forget to record the time when `push`? This may be a overkill though. I'd like to hear your thought.

```
type TimedTxQueue map[common.Hash]txAndTime

func (q *TimedTxQueue) push(tx) {}
func (q *TimedTxQueue) popLaterThan(timestamp) {}
func (q *TimedTxQueue) has(tx) {}
```

Reply:
I tried this but there were few functions which can be used repeatedly. I think this is overkill.

Reply:
https://github.com/kaiachain/kaia/pull/309/commits/8c8e51e9dc934631689f8fe55e9d266e95d56acf
https://github.com/kaiachain/kaia/pull/309/commits/277cdbe45301b292e446285932f95639973d6714